### PR TITLE
MAYA-115158 fix the python code to a a one-liner.

### PR DESCRIPTION
**Pull request by pierrebai-adsk**

### CODE REVIEW

File: plugin/adsk/scripts/mayaUsd_pluginUICreation.mel [hunk 1]
```diff
@@ -2,9 +2,16 @@ int $gHasPrimUpdater = -1;
...
     if ($gHasPrimUpdater < 0) {
-        $gHasPrimUpdater = `python("def checkPrimUp():\n  try:\n    from mayaUsd.lib import PrimUpdaterManager;\n    return 1;\n  except:\n    return 0;\ncheckPrimUp()")`;
+        // Note: we use a convoluted call to exec() in Python because
+        //       try block in Python cannot be one-liners and the MEL
+        //       python() function only returns the computed value when
+        //       the supplied Python code is a one-liner.
+        //
+        //       Putting the Python try block inside exec() allow us to
+        //       makethe Python code look like a one-liner to MEL.
+        $gHasPrimUpdater = `python("res = -1; exec('try:\\n    from mayaUsd.lib import PrimUpdaterManager;\\n    res = 1;\\nexcept:\\n    res = 0;\\n'); res")`;
```
Comment (pierrebai-adsk) [inline plugin/adsk/scripts/mayaUsd_pluginUICreation.mel:14]: Note: exec() uses the locals from the caller, so the res set within exec(0 is seen in the res variable outside it.

**Comment by pierrebai-adsk** [comment]
I tested this in 2022 and 2023.

File: plugin/adsk/scripts/mayaUsd_pluginUICreation.mel (hunk 1 above)
Comment (seando-adsk) [reply plugin/adsk/scripts/mayaUsd_pluginUICreation.mel:14]: In python 2.7, exec is a statement whereas in py3 its a function. Is this code expected to work in Maya 2022 when using py2?

**Comment by pierrebai-adsk** [comment]
Note: the only PF failure is the known vp2 isolate select test failure on Linux.

File: plugin/adsk/scripts/mayaUsd_pluginUICreation.mel (hunk 1 above)
Comment (pierrebai-adsk) [reply plugin/adsk/scripts/mayaUsd_pluginUICreation.mel:14]: But the calling convention is the same, isn't it? What would break?

Comment (pierrebai-adsk) [reply plugin/adsk/scripts/mayaUsd_pluginUICreation.mel:14]: I just tried it in a python 2.7 prompt and the code worked.

Comment (pierrebai-adsk) [reply plugin/adsk/scripts/mayaUsd_pluginUICreation.mel:14]: `>>> res = -1; exec('try:\n    from mayaUsd.lib import sys;\n    res = 1;\nexcept:\n    res = 0;\n'); print(res)
0
>>> res = -1; exec('try:\n    import sys;\n    res = 1;\nexcept:\n    res = 0;\n'); print(res)
1
>>>
`

Comment (seando-adsk) [reply plugin/adsk/scripts/mayaUsd_pluginUICreation.mel:14]: Excellent. If it works, then I have no concerns.

**Review by seando-adsk — APPROVED** [review]

**Pull request merged**